mobile/android/android-emulator-services.ts

```diff
@@ -120,15 +152,124 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 		return timeout * 1000;
 	}
 
+	private listVirtualMachines(parameter: string): IFuture<VirtualMachine[]> {
+		var future = new Future<VirtualMachine[]>();
+		var allData: string = "";
+		var virtualMachines: VirtualMachine[] = [];
+
+		var vboxPath = path.join(this.getPathToVBox().wait(), "vboxmanage");
+		var vboxManage = this.$childProcess.spawn(vboxPath, ["list", parameter]);
+
+		vboxManage.stdout.on("data", (chunk: string) => {
+			if(chunk) {
+				allData += chunk;
+			}
+		});
+		vboxManage.stderr.on("data", (error: Error) => {
+			// maybe use log trace level to print the error
+			this.$errors.fail("Unable to get currently running virtual machines. Check if VirtualBox is installed correctly.");
+		});
+
+		vboxManage.on("exit", () => {
+			_(allData.split(os.EOL)).each(row => {
+				// sample output row: "GN5" {1c268a7b-aa04-49c9-aabd-90595a747dec}
+				var rx = row.match(/^"(.+)"\s+\{(.+)\}$/);
+				if(rx && rx[1]) {
+					virtualMachines.push(new VirtualMachine(rx[1], rx[2]));
+				}
+			});
+
+			future.return(virtualMachines);
+		});
+
+		return future;
+	}
+
+	private shouldStartGenymotionEmulator(image: string): IFuture<boolean> {
+		return (() => {
+			if(!options.geny) {
+				return false;
+			}
+
+			var hasVmWithSpeciedImage = false;
+			var allVMs: VirtualMachine[] = this.listVirtualMachines("vms").wait();
+			_(allVMs).each((vm: VirtualMachine) => {
+				if(vm.name === image || vm.identifier === image) {
+					hasVmWithSpeciedImage = true;
+					// break each
+					return false;
+				}
+			});
+
+			if(!hasVmWithSpeciedImage) {
+				this.$errors.fail("There is no VirtualBox machine with name %s. Please verify specified name is correct.", image);
+			}
+
+			var shouldStartEmulator = true;
+			var virtualMachines: VirtualMachine[] = this.listVirtualMachines("runningvms").wait();
+			_(virtualMachines).each((vm: VirtualMachine) => {
+				if(vm.name === image || vm.identifier === image) {
+					shouldStartEmulator = false;
+					// break each
+					return false;
+				}
+			});
+
+			return shouldStartEmulator;
+		}).future<boolean>()();
+	}
+
+	private getPathToVBox(): IFuture<string> {
+		if(hostInfo.isWindows()) {
+			var result = new Future<string>();
+			var regKey64 = new Winreg({
+				hive: Winreg.HKLM,
+				key: '\\SOFTWARE\\Oracle\\VirtualBox',
+				regVersion: '64'
+			});
+
+			var regKey32 = new Winreg({
+				hive: Winreg.HKLM,
+				key: '\\SOFTWARE\\Oracle\\VirtualBox',
+				regVersion: '32'
+			});
+
+			regKey64.get("InstallDir", (err: Error, item: any) => {
```

Comment:
IMHO, it does not matter. On 64 bit OS you'll have 64-bit registration and on 32-bit OS - 32 bit registration.